### DISCUSSION TITLE
Klipper mcu firmware update

### DISCUSSION THREAD

**Discussion by Hayabusa454** [OpenNeptune3D/OpenNept4une#35] [Q&A] [answered]
I  de elegooed my N4P  about 2 months ago and everything is going great. A few days ago klipper release a new version and I updated it after a reboot received a mcu firmware error message. After a reboot error went away. My question is if i do go into boot mode and do make clean,make menucor,make which i believe compiles the new firmware after i flash new klipper mcu firmware do i lose amy of my configuration or not? Thanks John

**Comment by halfmanbear** [comment, marked as answer]
Hello, no you won't loose any configuration doing the above MCU update process, klippers latest release requires the MCU physical and MCU Virtual to be on a similar up-to-date version. The virtual MCU can be updated within the main OpenNept4une.sh installer. But the physical MCU you describe above is a more hands on processes.

You can always back up your printer.cfg worth doing regardless. It can be downloaded via fluidds config/logs tab

**Reply by Hayabusa454** [reply]
Thanks for the response and for doing this project for the community. Is it correct in saying I can start the process from "how to flash updated klipper mcu firmware" in the read me doc? Thanks again John

**Comment by halfmanbear** [comment]
No worries! Yes that's the doc the readme here

https://github.com/halfmanbear/OpenNept4une/tree/main/mcu-firmware

**Reply by steve8x8** [reply]
Stupid question: what would happen if there are two files in the root of the sd card, X_4.bin *and* elegoo_k1.bin? Would the printer find, use and rename the right one?

**Reply by halfmanbear** [reply]
Yes, having both on there is fine. Like you said it will look for one of the two then select it, flash & rename. I've updated the process to make this easier and suggesting to copy both variants to the microSD

**Reply by steve8x8** [reply]
Please copy the instructions for upgrading the (real) MCU to the wiki (and connect them with the "hidden SD" article already there - that one is currently the only place that hints at the changed procedure)

**Reply by halfmanbear** [reply]
I've made changes on the dev branch. Currently nowhere links to the old MCU readme so it's now not needed
